dotcom-rendering/src/components/Discussion/Comments.tsx

```diff
@@ -191,11 +189,11 @@ export const Comments = ({
 			commentCount / newFilterObject.pageSize,
 		);
 
-		if (page > maxPagePossible) setPage(maxPagePossible);
-
-		setFilters(newFilterObject);
-		// Filters also show when the view is not expanded but we want to expand when they're changed
-		onExpand();
+		if (page > maxPagePossible) {
+			handleFilterChange(filters, maxPagePossible);
+		} else {
+			handleFilterChange(filters);
+		}
```

Comment:
I wonder is this is the expected behaviour and whether we should simply remap the page instead. If I’m on page 5 with 25 comments per page I can see comments 125-150. If I move to 100 comments per page, I’d expect to be put on page 2, where I can see comments 100-200.

A [cross-multiplication](https://en.wikipedia.org/wiki/Cross-multiplication) should be sufficient:

```js
const desiredPage = Math.ceil(page * pageSize / newFilters.pageSize );
```

Reply:
Yep that sounds reasonable! I'll update this in a separate PR